Update HasGroups spec

niklassiemer opened the pull request.

niklassiemer committed:
[ci skip] lift non-None return for `__getitem__(node)`

pmrv commented on specs/has_groups.md:
```diff
@@ -12,7 +12,7 @@ also provide a way to actually access the contained objects.  This is done via `
 vertices.  From this we formulate the requirements for this interface
 
-1. `__getitem__()` must return a non-`None` object for each string returned by `list_nodes()` and `list_groups()`.
+1. `__getitem__()` must return any or a non-`None` object for each string returned by `list_nodes()` or `list_groups()`, repectively.
```
> must return any or a non-`None` object

Not sure what that means.

liamhuber replied:
"any of"?

niklassiemer replied:
Sorry,
'any object or a non-None type object' since getitem of a node might return `None`...

niklassiemer replied:
Is it better now?

niklassiemer committed:
[ci skip] list non-None object restriction.

The not-None restriction lifted for nodes and retained for groups, since they have to be instances of `HasGroups`.

niklassiemer commented:
I just realized we are violating our spec! `Project.list_nodes()` returns a list of jobs. However, any job isinstance of `HasGroups`!
